New structure for FAQs (#191)
seqeralabs/docs: platform_versioned_docs/version-24.1/troubleshooting_and_faqs/troubleshooting.mdx

---
title: "General troubleshooting"
description: "Troubleshooting Seqera Platform"
date: "24 Apr 2023"
tags: [troubleshooting, help]
---

## Common errors

**_Unknown pipeline repository or missing credentials_ error from public GitHub repositories**

GitHub imposes [rate limits](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting) on repository pulls (including public repositories), where unauthenticated requests are capped at 60 requests/hour and authenticated requests are capped at 5000 requests/hour. Seqera Platform users tend to encounter this error due to the 60 requests/hour cap.

Try the following:

1. Ensure there's at least one GitHub credential in your workspace's **Credentials** tab.

> 2. Ensure that the **Access token** field of all GitHub credential objects is populated with a [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) value and **not** a user password. GitHub PATs are typically longer than passwords and include a `ghp_` prefix. For example:`ghp*IqIMNOZH6zOwIEB4T9A2g4EHMy8Ji42q4HA`

Reviewer: ```suggestion
2. Ensure that the **Access token** field of all GitHub credential objects is populated with a [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) value and **not** a user password. GitHub PATs are typically longer than passwords and include a `ghp_` prefix. For example: `ghp*IqIMNOZH6zOwIEB4T9A2g4EHMy8Ji42q4HA`
```

To fix the problem, try the following:

1. Set a JDK environment variable to force Nextflow and Seqera containers to use TLSv1.2 by default:

    ```
    export JAVA_OPTIONS="-Dmail.smtp.ssl.protocols=TLSv1.2"
    ```

2. Add this parameter to your [nextflow.config file](./launch/advanced.mdx#nextflow-config-file):

    ```
    mail {
        smtp.ssl.protocols = 'TLSv1.2'
    }
    ```

3. Ensure these values are also set for Nextflow and/or Seqera:

    -   `mail.smtp.starttls.enable=true`
    -   `mail.smtp.starttls.required=true`

## Git integration

**BitBucket authentication failure: _Can't retrieve revisions for pipeline - https://my.bitbucketserver.com/path/to/pipeline/repo - Cause: Get branches operation not supported by BitbucketServerRepositoryProvider provider_**

If you supplied the correct BitBucket credentials and URL details in your `tower.yml` and still experience this error, update your version to at least v22.3.0. This version addresses SCM provider authentication issues and is likely to resolve the retrieval failure described here.

## Healthcheck

**Seqera Platform API healthcheck endpoint**

> To implement automated healthcheck functionality, use Seqera's `service-info` endpoint.
> 
> Example:

Reviewer: ```suggestion
To implement automated healthcheck functionality, use Seqera's `service-info` endpoint. For example:
```

Improvements are being made to the way Nextflow calculates the optimal memory needed for containerized tasks, which will resolve issues with underestimating memory allocation in an upcoming release.

A temporary workaround for this issue is to implement a `retry` error strategy in the failing process that will increase the allocated memory each time the failed task is retried. Add the following `errorStrategy` block to the failing process:

```bash
process {
    errorStrategy = 'retry'
    maxRetries    = 3
    memory = { 1.GB * task.attempt }
}
```

## Plugins

**Use the Nextflow SQL DB plugin to query AWS Athena**

From [Nextflow 22.05.0-edge](https://github.com/nextflow-io/nextflow/releases/tag/v22.05.0-edge), your Nextflow pipelines can query data from AWS Athena. Add these configuration items to your `nextflow.config`. The use of secrets is optional:

```
plugins {
  id 'nf-sqldb@0.4.0'
}

sql {
    db {
        'athena' {
              url = 'jdbc:awsathena://AwsRegion=YOUR_REGION;S3OutputLocation=s3://YOUR_S3_BUCKET'
              user = secrets.ATHENA_USER
              password = secrets.ATHENA_PASSWORD
            }
    }
}
```

> Then, call the functionality in your workflow:
> 
> ```
> // Example
>   channel.sql.fromQuery("select * from test", db: "athena", emitColumns:true).view()
> }
> ```

Reviewer: ```suggestion
Then, call the functionality in your workflow. For example:

```
  channel.sql.fromQuery("select * from test", db: "athena", emitColumns:true).view()
}
```
```
